lib/trip.rb

```diff
@@ -46,22 +76,26 @@ def inspect
         "PassengerID=#{passenger&.id.inspect}>"
     end
 
-    def connect(passenger)
+    def connect(passenger,driver)
       @passenger = passenger
+      @driver = driver
       passenger.add_trip(self)
+      driver.add_trip(self)
     end
 
     private
 
     def self.from_csv(record)
       return self.new(
                id: record[:id],
+               driver_id: record[:driver_id],
                passenger_id: record[:passenger_id],
-               start_time: record[:start_time],
-               end_time: record[:end_time],
+               start_time: record[:start_time], # nn to turn into Time class
+               end_time: record[:end_time], # nn to turn into Time class  before passing into initialize 
```

Comment:
You should use `Time.parse` here:

```suggestion
               start_time: Time.parse(record[:start_time]),
               end_time: Time.parse(record[:end_time]),
```